File: movr/lab_json/movr_java_json/src/main/java/io/roach/movrapi/entity/LocationHistory.java
package io.roach.movrapi.entity;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.UUID;

/**
 * Hibernate entity for Vehicles' Location History Table
 */

@Entity
@Table(name = "location_history")
public class LocationHistory {

    @Id
    @GeneratedValue
    private UUID id;
    @Column(name = "ts")
    private Timestamp timestamp;
    private Double longitude;
    private Double latitude;
    @ManyToOne
    @JoinColumn(name = "vehicle_id", nullable = false)
    private Vehicle vehicle;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public Timestamp getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(Timestamp timestamp) {
        this.timestamp = timestamp;
    }

    public Double getLongitude() {
        return longitude;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }

    public Double getLatitude() {
        return latitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public Vehicle getVehicle() {
        return vehicle;
    }

    public void setVehicle(Vehicle vehicle) {
        this.vehicle = vehicle;
    }
}

File: movr/lab_complete/movr_node_complete/src/entity/LocationHistory.ts
import {Entity, PrimaryColumn, Column, AfterLoad } from "typeorm";
import * as moment from "moment"

@Entity({
    name:"location_history"
})
export class LocationHistory {

    @PrimaryColumn("uuid")
    id: string;

    @Column("uuid")
    vehicle_id: string;

    @Column({
        type:"timestamp"
    })
    ts: string;

    @Column({ type: 'decimal', precision: 5, scale: 1})
    longitude: number;

    @Column({ type: 'decimal', precision: 5, scale: 1})
    latitude: number;

    @AfterLoad()
    formatTime(){
        const pattern = 'YYYY-MM-DD HH:mm:ss'
        this.ts = moment(this.ts).format(pattern)
    }
}